tests/integration/plugins/test_npm.py

```diff
@@ -166,3 +183,69 @@ def test_npm_plugin_include_node(new_dir, partitions):
     # try to use bundled Node.js to execute the script
     output = subprocess.check_output([str(node_path), str(binary)], text=True)
     assert output == "hello world\n"
+
+
+@pytest.mark.slow
+def test_npm_plugin_get_file_list(create_fake_package_with_node, new_dir, partitions):
+    parts = create_fake_package_with_node()
+    lifecycle = LifecycleManager(
+        parts,
+        application_name="test_npm_plugin",
+        cache_dir=new_dir,
+        partitions=partitions,
+    )
+    actions = lifecycle.plan(Step.BUILD)
+
+    with lifecycle.action_executor() as ctx:
+        ctx.execute(actions)
+
+    part_name = list(parts["parts"].keys())[0]
+    actual_file_list = lifecycle._executor._handler[part_name]._plugin.get_file_list()
+
+    # This example bundles in node, which brings a ton of other dependencies -
+    # this is perfect for checking all sorts of weird behavior.
+
+    for (pkg_name, _), pkg_files in actual_file_list.items():
+        # Check for the files from our little fake package
+        if pkg_name == "npm-hello":
+            pkg_files_rerooted = _make_paths_relative(pkg_files)
+            assert pkg_files_rerooted == {
+                "bin/npm-hello",
+                "lib/node_modules/npm-hello/hello.js",
+                "lib/node_modules/npm-hello/package.json",
+            }
+
+        # Verify bins were installed properly
+        if pkg_name == "npm":
+            pkg_files_rerooted = _make_paths_relative(pkg_files)
+            assert "bin/npm" in pkg_files_rerooted
+            assert "bin/npx" in pkg_files_rerooted
+
+    # Node itself depends on four different versions of this ansi-regex
+    # package. npm itself directly depends on 2.1.1, and two of npm's
+    # dependencies (cli-columns and gauge) both depend on 5.0.1, which means
+    # their files get collapsed under a single key (which doesn't matter for
+    # our purposes.)
+    ar211 = ("ansi-regex", "2.1.1")
+    assert ar211 in actual_file_list
+    assert len(actual_file_list[ar211]) == 4, ar211
+
+    ar300 = ("ansi-regex", "3.0.0")
+    assert ar300 in actual_file_list
+    assert len(actual_file_list[ar300]) == 4, ar300
+
+    # Added "index.d.ts" file, absent from previous versions
+    ar500 = ("ansi-regex", "5.0.0")
+    assert ar500 in actual_file_list
+    assert len(actual_file_list[ar500]) == 5, ar500
+
+    # Between 5.0.0 and 5.0.1 they seem to have stopped packaging the readme;
+    # back to 4 files per install.
```

Comment:
but the assertion says `8`?

Reply:
Comment clarified.

Reply:
https://github.com/canonical/craft-parts/pull/931/commits/579fd4c638648e50eecd4814e51e2974871182c7